packages/dev/addons/src/navigation/factory/factory.worker.ts

```diff
@@ -1,55 +1,55 @@
-//         Logger.Warn("createNavMesh is not supported in worker mode, use createNavMeshAsync instead.");
-//         return null;
-//     };
-
-//     plugin.createNavMeshAsync = async (meshes: Mesh[], parameters: INavMeshParametersV2) => {
-//         return await new Promise<{
-//             navMesh: NavMesh;
-//             navMeshQuery: NavMeshQuery;
-//             tileCache?: TileCache;
-//         }>((resolve, _reject) => {
-//             GenerateNavMeshWithWorker(meshes, parameters, {
-//                 worker,
-//                 completion: (navMesh: NavMesh, navMeshQuery: NavMeshQuery, tileCache?: TileCache) => {
-//                     resolve({ navMesh, navMeshQuery, tileCache });
-//                 },
-//             });
-//         });
-//     };
-
-//     return plugin;
-// }
+import type { NavMesh, NavMeshQuery, TileCache } from "@recast-navigation/core";
+import { init as initRecast } from "@recast-navigation/core";
+
+import { Logger } from "core/Misc/logger";
+import type { Mesh } from "core/Meshes/mesh";
+
+import { RecastNavigationJSPluginV2 } from "../plugin/RecastNavigationJSPlugin";
+import type { INavMeshParametersV2 } from "../types";
+import { GenerateNavMeshWithWorker } from "../generator/generator.worker";
+import { CreateNavigationPluginAsync } from "./factory.single-thread";
+import { GenerateNavMeshWorker } from "../worker/navmesh-worker";
+
+/**
+ * Creates a navigation plugin for the given scene using a worker.
+ * @returns A promise that resolves to the created navigation plugin.
+ * @remarks This function initializes the Recast module and sets up the navigation plugin to use a worker.
+ * The worker is used to handle the creation of the navigation mesh asynchronously.
+ * The `createNavMesh` method is not supported in worker mode, use `createNavMeshAsync` instead.
+ */
+export async function CreateNavigationPluginWorkerAsync() {
+    if (window && !window.Worker) {
+        Logger.Error("Web Workers are not supported in this environment. Please ensure your environment supports Web Workers.");
+        return await CreateNavigationPluginAsync(); // Fallback to single-threaded version
+    }
+
+    await initRecast();
+
+    const plugin = new RecastNavigationJSPluginV2();
+
+    // Convert the worker function to a blob
+    const workerBlob = new Blob([`(${GenerateNavMeshWorker.toString()})()`], { type: "application/javascript" });
+    const worker = new Worker(URL.createObjectURL(workerBlob));
+
+    plugin.createNavMesh = () => {
+        Logger.Warn("createNavMesh is not supported in worker mode, use createNavMeshAsync instead.");
+        return null;
+    };
+
+    plugin.createNavMeshAsync = async (meshes: Mesh[], parameters: INavMeshParametersV2) => {
+        return await new Promise<{
+            navMesh: NavMesh;
+            navMeshQuery: NavMeshQuery;
+            tileCache?: TileCache;
+        }>((resolve, _reject) => {
+            GenerateNavMeshWithWorker(meshes, parameters, {
```

Comment:
Seems like `GenerateNavMeshWithWorker` can synchronously throw, but there is no try/catch with a call to _reject. I think that should be added if I'm reading the code correctly.

Reply:
Do not review the worker please yet.

Reply:
Where are we at with this? On the one hand, it seems you are saying "please don't review this," but on the other hand, it is part of the PR that will be merged. So ready for review or no?

Reply:
RaananW was supposed the help with this one... I really don't want to remove the files (they will do no harm at all)  so I don't get into a git hell when there will be an opportunity to solve the issue.

https://forum.babylonjs.com/t/replacing-recastjs-with-recast-navigation-js/56003/46

Reply:
If that is the case, then can you put a comment at the top of the file explaining that there are outstanding issues in this code that need to be fixed, and what those issues are? Otherwise it looks like this is expected to be working code.

Reply:
Also, if you explain what problems you are having with the workers @VicenteCartas might be able to help as well.

Reply:
See the previous link I sent you. Thanks!